### PR TITLE
chore: upgrade to polkadot-v1.1.0

### PR DESCRIPTION
**Pull request by Roznovjak**
TODO:
- [x] use XYK and LBP pallets from hydra
- [x] rebenchmark all pallets
- [x] storage migrations
- [x] test runtime upgrade (try-runtime and chopsticks)
- [x] update main README file

Our stuff:
- [x] XCMP defer queue
- [x] XCM executor atomicity
- [x] Tinkernet
- [x] Democracy pallet
- [x] rust 1.71 compatibily with Zombienet (`UnknownOpcode(192)`)

Notable changes:
- Deprecate `Weight::from_ref_time` and `Weight::from_proof_size` and use `from_parts` instead
- The `try-runtime` subcommand has been migrated to a standalone CLI [#14731](https://github.com/paritytech/substrate/pull/14731)
- [XCM Topics](https://github.com/paritytech/polkadot/pull/7234) - XCM messages are uniquely referenceable
- `BlockNumber` removed from `frame_system::Config`. Use `frame_system::BlockNumberFor<T>`. #[14437](https://github.com/paritytech/substrate/pull/14437)
- Default config parameters #[13454](https://github.com/paritytech/substrate/pull/13454)

API changes:
- pallet_balances #[12951](https://github.com/paritytech/substrate/pull/12951)
  - `set_balance` becomes `force_set_balance`
  - `transfer` becomes `transfer_allow_death`

This release removes the following CLI parameters:

    --rpc-max-payload (replaced by --rpc-max-request-size and --rpc-max-response-size)
    --ws-max-out-buffer-capacity
    --ws-external (replaced by --rpc-external)
    --unsafe-ws-external (replaced by --unsafe-rpc-external)
    --ipc--path
    --ws-port (replaced by --rpc-port)
    --ws-max--connections (replaced by --rpc-max-connections)
    --rpc-http (replaced by --rpc-addr)
    --rpc-ws (replaced by --rpc-addr)

### CODE REVIEW

File: runtime/basilisk/src/assets.rs [hunk 1]
```diff
@@ -246,6 +268,9 @@ impl pallet_lbp::Config for Runtime {
 	type BlockNumberProvider = RelayChainBlockNumberProvider<Runtime>;
 }
 
+#[cfg(feature = "runtime-benchmarks")]
```
Comment (mrq1911) [inline runtime/basilisk/src/assets.rs:271]: move to the top

File: runtime/basilisk/src/assets.rs [hunk 2]
```diff
@@ -381,6 +406,40 @@ impl warehouse_liquidity_mining::Config<XYKLiquidityMiningInstance> for Runtime
 	type PriceAdjustment = warehouse_liquidity_mining::DefaultPriceAdjustment;
 }
 
+// TODO
```
Comment (mrq1911) [inline runtime/basilisk/src/assets.rs:409]: should be done?

File: runtime/basilisk/src/governance.rs [hunk 3]
```diff
@@ -149,7 +154,9 @@ impl pallet_collective::Config<CouncilCollective> for Runtime {
 	type MaxProposals = CouncilMaxProposals;
 	type MaxMembers = CouncilMaxMembers;
 	type DefaultVote = pallet_collective::PrimeDefaultVote;
-	type WeightInfo = ();
+	type WeightInfo = (); // TODO
```
Comment (mrq1911) [inline runtime/basilisk/src/governance.rs:157]: this definetly should be done

File: runtime/basilisk/src/xcm.rs [hunk 4]
```diff
@@ -123,10 +122,14 @@ impl WeightTrader for TradePassthrough {
...
 	}
 
-	fn buy_weight(&mut self, _weight: Weight, payment: Assets) -> Result<Assets, Error> {
+	fn buy_weight(&mut self, _weight: Weight, payment: Assets, _context: &XcmContext) -> Result<Assets, Error> {
 		// Just let it through for now
 		Ok(payment)
 	}
+
+	fn refund_weight(&mut self, _weight: Weight, _context: &XcmContext) -> Option<MultiAsset> {
+		todo!()
```
Comment (mrq1911) [inline runtime/basilisk/src/xcm.rs:131]: should be implemented?

File: runtime/basilisk/src/xcm.rs [hunk 5]
```diff
@@ -268,6 +273,9 @@ impl pallet_xcm::Config for Runtime {
 	type WeightInfo = weights::xcm::BasiliskWeight<Runtime>;
 	#[cfg(feature = "runtime-benchmarks")]
 	type ReachableDest = ReachableDest;
+	type AdminOrigin = SuperMajorityCouncilOrRoot;
```
Comment (mrq1911) [inline runtime/basilisk/src/xcm.rs:276]: we probably dont want council to have this privilege

File: runtime/basilisk/src/xcm.rs [hunk 6]
```diff
@@ -268,6 +273,9 @@ impl pallet_xcm::Config for Runtime {
...
+	type MaxRemoteLockConsumers = ConstU32<0>;
```
Comment (mrq1911) [inline runtime/basilisk/src/xcm.rs:277]: why is this 0?

**Review by mrq1911 — CHANGES_REQUESTED** [review]

File: node/src/cli.rs [hunk 7]
```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::large_enum_variant)]
```
[the remark this answers is not in the document]
Comment (Roznovjak) [reply node/src/cli.rs:1]: I think so. It looks like there is no version of it that's meant to be reused by parachains.

File: runtime/basilisk/src/governance.rs (hunk 3 above)
Comment (Roznovjak) [reply runtime/basilisk/src/governance.rs:157]: btw, there is more pallets that are configured to use default weights

File: runtime/basilisk/src/governance.rs [hunk 8]
```diff
@@ -167,7 +174,9 @@ impl pallet_collective::Config<TechnicalCollective> for Runtime {
 	type MaxProposals = TechnicalMaxProposals;
 	type MaxMembers = TechnicalMaxMembers;
 	type DefaultVote = pallet_collective::PrimeDefaultVote;
-	type WeightInfo = ();
+	type WeightInfo = (); // TODO
```
Comment (dmoka) [inline runtime/basilisk/src/governance.rs:177]: same here, missing TODO

File: runtime/basilisk/src/xcm.rs (hunk 5 above)
Comment (Roznovjak) [reply runtime/basilisk/src/xcm.rs:260]: SuperMajorityTechCommitteeOrRoot ?

**Review by mrq1911 — APPROVED** [review]

**Pull request merged**